osquery/CMakeLists.txt

```diff
@@ -36,11 +36,23 @@ set(OSQUERY_LIBS
   ${OPENSSL_SSL_LIBRARY}
   ${READLINE_LIBRARIES}
 
-  pthread
-  bz2
-  z
 )
 
+if(WIN32)
+  set(OSQUERY_LIBS 
+    ${OSQUERY_LIBS} 
+    ${BZIP2_LIBRARIES}
+    ${boost_thread_library}   
+  )
+else()
+  set(OSQUERY_LIBS 
+    ${OSQUERY_LIBS} 
+    z 
```

Comment:
Why not create a CMake platform conditional that populates `${BZIP2_LIBRARIES}` on POSIX, as well as the threading implementation, then have one append to `${OSQUERY_LIBS}` here.